File: GoLang/剑指Offer/easy/剑指Offer_53_I在排序数组中查找数字I.go
package easy

func search(nums []int, target int) int {
	if nums == nil || len(nums) == 0 {
		return 0
	}
	left, right := 0, len(nums)-1

	for left <= right && nums[left] != target {
		mid := (left + right) / 2
		if nums[mid] > target {
			right = mid - 1
		} else if nums[mid] < target {
			left = mid + 1
		} else {
			left = mid
			break
		}
	}

	if left >= len(nums) || nums[left] != target {
		return 0
	}
	l, r := left, left
	for l >= 0 && nums[l] == target {
		l--
	}
	for r < len(nums) && nums[r] == target {
		r++
	}
	return r - l - 1
}

File: Java/剑指Offer/easy/剑指Offer_53_I在排序数组中查找数字I.java
package easy;

/**
 * Created by lwd at 2021/8/5
 *
 * @Description:
 */
public class 剑指Offer_53_I在排序数组中查找数字I {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length - 1, mid = 0;
        while (left < right) {
            mid = (left + right) / 2;
            if (nums[mid] == target) break;
            else if (nums[mid] > target) right = mid - 1;
            else left = mid + 1;
        }

        if (nums[mid] != target) return 0;
        left = mid;
        right = mid;
        while (left >= 0 && nums[left] == target) left--;
        while (right < nums.length && nums[right] == target) right++;
        return right - left - 1;
    }
}
